pepper-apis/src/main/java/org/broadinstitute/ddp/route/GovernedParticipantRegistrationRoute.java

```diff
@@ -0,0 +1,100 @@
+package org.broadinstitute.ddp.route;
+
+import java.time.ZoneId;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.broadinstitute.ddp.constants.ErrorCodes;
+import org.broadinstitute.ddp.constants.RouteConstants;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.JdbiUmbrellaStudyCached;
+import org.broadinstitute.ddp.db.dao.JdbiUserStudyEnrollment;
+import org.broadinstitute.ddp.db.dao.StudyGovernanceDao;
+import org.broadinstitute.ddp.db.dao.UserDao;
+import org.broadinstitute.ddp.db.dao.UserGovernanceDao;
+import org.broadinstitute.ddp.db.dao.UserProfileDao;
+import org.broadinstitute.ddp.db.dto.LanguageDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.json.GovernedUserRegistrationPayload;
+import org.broadinstitute.ddp.json.UserRegistrationResponse;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.model.governance.Governance;
+import org.broadinstitute.ddp.model.governance.GovernancePolicy;
+import org.broadinstitute.ddp.model.user.EnrollmentStatusType;
+import org.broadinstitute.ddp.model.user.User;
+import org.broadinstitute.ddp.model.user.UserProfile;
+import org.broadinstitute.ddp.util.DateTimeUtils;
+import org.broadinstitute.ddp.util.I18nUtil;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import org.broadinstitute.ddp.util.ValidatedJsonInputRoute;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+public class GovernedParticipantRegistrationRoute extends ValidatedJsonInputRoute<GovernedUserRegistrationPayload> {
+    private static final Logger LOG = LoggerFactory.getLogger(GovernedParticipantRegistrationRoute.class);
+
+    @Override
+    public Object handle(Request request, Response response, GovernedUserRegistrationPayload payload) throws Exception {
+        String operatorGuid = request.params(RouteConstants.PathParam.USER_GUID);
+        String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
+        return TransactionWrapper.withTxn(handle -> {
+            UserGovernanceDao userGovernanceDao = handle.attach(UserGovernanceDao.class);
+            UserDao userDao = handle.attach(UserDao.class);
+            Optional<Long> studyId = new JdbiUmbrellaStudyCached(handle).getIdByGuid(studyGuid);
+            if (studyId.isEmpty()) {
+                ApiError error = new ApiError(ErrorCodes.STUDY_NOT_FOUND, "Study " + studyGuid + " does not exist");
+                throw ResponseUtil.haltError(response, HttpStatus.SC_NOT_FOUND, error);
+            }
+            User operatorUser = userDao.findUserByGuid(operatorGuid)
+                    .orElseThrow(() -> new DDPException("Could not find operator with guid " + operatorGuid));
+            long auth0ClientId = operatorUser.getCreatedByClientId();
```

Comment:
I think it will be better to get the client from `DDPAuth`, e.g. the client who is making the API call.

Reply:
Fixed

Reply:
Didn't see any changes related to this, did we forget to push a commit?

Reply:
Now fixed

Reply:
Very nice!